polkadot/zombienet-sdk-tests/tests/functional/approval_voting_coalescing.rs

```diff
@@ -0,0 +1,192 @@
+				.with_collator(|n| n.with_name("collator-undying-2003"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2004)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2004"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2005)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2005"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2006)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2006"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2007)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2007"))
+		})
+		.build()
+		.map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})?;
+
+	let spawn_fn = zombienet_sdk::environment::get_spawn_fn();
+
+	log::info!("Spawning network");
+	let network = spawn_fn(config).await?;
+
+	log::info!("Waiting for network to initialize");
+	let relay_node = network.get_node("validator-0")?;
+	let para_node_2001 = network.get_node("collator-undying-2000")?;
+
+	let relay_client: OnlineClient<PolkadotConfig> = relay_node.wait_client().await?;
+
+	log::info!("Waiting for parachains to advance to block 15");
+	assert_para_throughput(
+		&relay_client,
+		15,
+		[
+			(ParaId::from(2000), 11..16),
+			(ParaId::from(2001), 11..16),
+			(ParaId::from(2002), 11..16),
+			(ParaId::from(2003), 11..16),
+			(ParaId::from(2004), 11..16),
+			(ParaId::from(2005), 11..16),
+			(ParaId::from(2006), 11..16),
+			(ParaId::from(2007), 11..16),
+		]
+		.into_iter()
+		.collect(),
+	)
+	.await?;
+
+	log::info!("Checking finality does not log and no-shows are within range");
+	for node in network.nodes() {
+		assert_finality_lag_less_than(&node.wait_client().await?, no_show_slots).await?;
+		assert!(
+			node.reports("polkadot_parachain_approvals_no_shows_total").await.unwrap() < 3.1,
+			"No-shows should be less than 3"
```

Comment:
This is confusing, above we compare < 3.1 ? which is <=3.

Reply:
Yeah brain fart on my end. Fixing it.